### ISSUE TITLE
Question: Use with Typescript

### ISSUE THREAD

**Issue by bradjones1** [exivity/react-orbitjs#260]
Thanks again for this great little package. I'm trying to venture more into TypeScript, but I'm getting an inspection in my IDE that has me a bit stumped.

```
SendIntroduction = withData()(SendIntroduction);

Assigned expression type ConnectedComponentClass<(props: (NativeStackScreenProps<{datingProfile: InitializedRecord}> & WithUpdatableDataProps)) => JSX.Element, Omit<GetProps<(props: (NativeStackScreenProps<{datingProfile: InitializedRecord}> & WithUpdatableDataProps)) => JSX.Element>, keyof {queryStore: (queryOrExpression: QueryOrExpression, options?: object, id?: string) => any, updateStore: (queryOrExpression: QueryOrExpression, options?: object, id?: string) => any} & WithData>> is not assignable to type (props: (NativeStackScreenProps<{datingProfile: InitializedRecord}> & WithUpdatableDataProps)) => JSX.Element 
```

`SendIntroduction` is a functional component. I can ignore this with an inline hint for PhpStorm, but I'm curious what I'm missing?

**Comment by hongaar** [comment]
I'm afraid I can't help with this error. If you find a solution, I'd be happy to merge a PR to fix this!

**Comment by bradjones1** [comment]
So this is my error, though it wasn't super obvious to this TypeScript newbie. The returned component (the return value of `withData`...) is a new HOC and cannot be used to overwrite a function component of the same name. So if you want your component to be named `TheComponent`, have it created from something like `WrappedTheComponent`. Otherwise TypeScript will rightly say that the types do not match. Your code will compile and work just fine, but you'll have invalid TS in your IDE.

**Comment by hongaar** [comment]
TypeScript errors can be somewhat hard to parse language-wise in my experience, glad you found the problem though!
